androidsdk/src/test/java/com/wootric/androidsdk/WootricTest.java

```diff
@@ -115,6 +115,20 @@ public void setUp() {
         assertThat(wootric.endUser.getEmailOrUnknown()).isEqualTo("nps@example.com");
     }
 
+    @Test public void setEndUserExternalId() throws Exception {
```

Comment:
Then you should validate that at this level. I don't think  `" "`  is a valid ID